### DISCUSSION TITLE
Using ZLUDA with koboldcpp

### DISCUSSION THREAD

**Discussion by brknsoul** [LostRuins/koboldcpp#961] [General]
Would this be possible?

[ZLUDA](https://github.com/vosen/ZLUDA) is a CUDA wrapper for AMD GPUs.

[lshyqqtiger](https://github.com/lshqqytiger/) has got ZLUDA to work with various Stable Diffusion front ends ([A1111](https://github.com/lshqqytiger/stable-diffusion-webui-amdgpu), [Forge](https://github.com/lshqqytiger/stable-diffusion-webui-amdgpu-forge), [SD.Next](https://github.com/vladmandic/automatic)).

**Comment by LostRuins** [comment]
I don't have an AMD GPU. I'm not sure if the current GGML implementation using HIPBLAS works for stable diffusion, maybe you can check with @YellowRoseCx

**Comment by EmanuelOverride** [comment]
I got it to work after a bit of faff. https://github.com/YellowRoseCx/koboldcpp-rocm/discussions/34#discussioncomment-10184274

It looks like it could be added as an option without too much hassle, although it wouldn't run out of the box without installing HIP/Zluda/adding paths etc. But, if I can get it working, it's basically free.

**Reply by LioanZluda** [reply]
how did you get it to work?

**Reply by brknsoul** [reply]
Don't bother. Either use the ROCm [fork](https://github.com/YellowRoseCx/koboldcpp-rocm/), or use the no-cuda [release](https://github.com/LostRuins/koboldcpp/releases/tag/v1.95.1) with Vulkan backend.
